instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/ClosuresTest.kt

```diff
@@ -0,0 +1,238 @@
+import com.mapbox.navigation.instrumentation_tests.utils.readRawFileText
+import com.mapbox.navigation.testing.ui.BaseCoreNoCleanUpTest
+import com.mapbox.navigation.testing.ui.utils.MapboxNavigationRule
+import com.mapbox.navigation.testing.ui.utils.coroutines.getSuccessfulResultOrThrowException
+import com.mapbox.navigation.testing.ui.utils.coroutines.requestRoutes
+import com.mapbox.navigation.testing.ui.utils.coroutines.routeProgressUpdates
+import com.mapbox.navigation.testing.ui.utils.coroutines.routesUpdates
+import com.mapbox.navigation.testing.ui.utils.coroutines.sdkTest
+import com.mapbox.navigation.testing.ui.utils.coroutines.setNavigationRoutesAndWaitForUpdate
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import com.mapbox.navigation.testing.ui.utils.runOnMainSync
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.flow.toList
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.net.URI
+import kotlin.math.abs
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class ClosuresTest : BaseCoreNoCleanUpTest() {
+
+    @get:Rule
+    val mapboxNavigationRule = MapboxNavigationRule()
+
+    @get:Rule
+    val mockLocationReplayerRule = MockLocationReplayerRule(mockLocationUpdatesRule)
+
+    private val origin = Point.fromLngLat(-121.496066, 38.577764)
+    private val destination = Point.fromLngLat(-121.480256, 38.576795)
+
+    private lateinit var mapboxNavigation: MapboxNavigation
+
+    override fun setupMockLocation(): Location = mockLocationUpdatesRule.generateLocationUpdate {
+        latitude = origin.latitude()
+        longitude = origin.longitude()
+    }
+
+    @Before
+    fun setUp() {
+        runOnMainSync {
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(context)
+                    .accessToken(getMapboxAccessTokenFromResources(context))
+                    .routingTilesOptions(
+                        RoutingTilesOptions.Builder()
+                            .tilesBaseUri(URI(mockWebServerRule.baseUrl))
+                            .build()
+                    )
+                    .build()
+            )
+        }
+    }
+
+    @Test
+    fun closuresAppearedAfterRefresh() = sdkTest {
+        // no closures in the primary route
+        setUpRequestHandler(R.raw.route_response_route_refresh)
+        mockWebServerRule.requestHandlers.add(
+            FailByRequestMockRequestHandler(
+                MockDirectionsRefreshHandler(
+                    "route_response_route_refresh",
+                    // [1, 3] closure on leg#0
+                    readRawFileText(context, R.raw.route_response_route_refresh_annotations),
+                    acceptedGeometryIndex = 0
+                )
+            )
+        )
```

Comment:
Do you need `FailByRequestMockRequestHandler` wrapper?
```suggestion
        mockWebServerRule.requestHandlers.add(
                MockDirectionsRefreshHandler(
                    "route_response_route_refresh",
                    // [1, 3] closure on leg#0
                    readRawFileText(context, R.raw.route_response_route_refresh_annotations),
                    acceptedGeometryIndex = 0
                )
        )
```

Reply:
No really, I just copied it from some other test.